### PR TITLE
[INF-5307] - Add the MongoDB & PostgreSQL Outbox Ably Ingress Rules

### PR DESCRIPTION
**Pull request by graham-russell**
This adds the new Ably Ingress rule for MongoDB.

## Summary by CodeRabbit

- **New Features**
	- Introduced documentation for the `ably_ingress_rule_mongodb` resource.
	- Added documentation for the `ably_ingress_rule_postgres_outbox` resource.
	- Added new resources for managing MongoDB and PostgreSQL ingress rules.

- **Bug Fixes**
	- Updated `azure_app_id` field in multiple resources for consistency.

- **Documentation**
	- Enhanced documentation for the Ably Terraform Provider, including resource management capabilities and authentication instructions.
	- Updated example configurations for Azure function resources.

- **Tests**
	- Added acceptance tests for the `ably_ingress_rule_mongodb` and `ably_ingress_rule_postgres_outbox` resources.
	- Updated tests for the `ably_rule_azure_function` resource.
	- Added new variables for configuring MongoDB and PostgreSQL settings.

- **Chores**
	- Updated Go version matrix in the GitHub workflow for testing.
	- Updated provider source and version in example configurations.
	- Updated copyright year in the COPYRIGHT file.

### CODE REVIEW

**Comment by graham-russell** [comment]
@coderabbitai review

**Comment by graham-russell** [comment]
@coderabbitai review

**Comment by graham-russell** [comment]
@coderabbitai review

File: internal/provider/resource_ably_ingress_rule_mongo.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			"full_document_before_change": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "Controls wheter the full document before the change should be included in the change event. Full Document before change is not available on all types of change event. Possible values are whenAvailable or off. The default is off.",
+			},
+			"primary_site": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The primary site that the connector will run in. You should choose a site that is close to your database.",
+			},
```
[the remark this answers is not in the document]
Comment (graham-russell) [reply internal/provider/resource_ably_ingress_rule_mongo.go:54]: This needs an upgrade of tfsdk to be actioned.

File: internal/provider/resource_ably_ingress_rule_postgres_outbox.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+				Type:     types.StringType,
+				Required: true,
+				Description: `Determines the level of protection provided by the SSL connection. Options are:
+  - prefer: Attempt SSL but allow non-SSL.
+  - require: Always use SSL but don't verify certificates.
+  - verify-ca: Verify server certificate is signed by a trusted CA.
+  - verify-full: Verify server certificate and hostname.
+
+Default: prefer.`,
+			},
```
[the remark this answers is not in the document]
Comment (graham-russell) [reply internal/provider/resource_ably_ingress_rule_postgres_outbox.go:55]: Validators are not available in the current version of tfsdk in use.

File: internal/provider/resource_ably_ingress_rule_postgres_outbox.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+// Get Rule Resource schema
+func (r resourceIngressRulePostgresOutbox) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return GetIngressRuleSchema(
+		map[string]tfsdk.Attribute{
+			"url": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The URL for your Postgres database, for example postgres://user:password@example.com:5432/your-database-name. The associated user must have the correct privileges",
+			},
```
[the remark this answers is not in the document]
Comment (graham-russell) [reply internal/provider/resource_ably_ingress_rule_postgres_outbox.go:24]: this isn't a real password. It if for illustrative purposes.

**Review by emptyhammond — APPROVED** [review]

**Pull request merged**